discovery/digitalocean/digitalocean_test.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package digitalocean
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"testing"
+
+	"github.com/go-kit/kit/log"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/util/testutil"
+)
+
+type DigitalOceanSDTestSuite struct {
+	Mock *SDMock
+}
+
+func (s *DigitalOceanSDTestSuite) TearDownSuite() {
+	s.Mock.ShutdownServer()
+}
+
+func (s *DigitalOceanSDTestSuite) SetupTest(t *testing.T) {
+	s.Mock = NewSDMock(t)
+	s.Mock.Setup()
+
+	s.Mock.HandleDropletsList()
+}
+
+func TestDigitalOceanSDRefresh(t *testing.T) {
+	sdmock := &DigitalOceanSDTestSuite{}
+	sdmock.SetupTest(t)
+	t.Cleanup(sdmock.TearDownSuite)
+
+	cfg := DefaultSDConfig
+	cfg.BearerToken = tokenID
+	d, err := NewDiscovery(&cfg, log.NewNopLogger())
+	testutil.Ok(t, err)
+	endpoint, err := url.Parse(sdmock.Mock.Endpoint())
+	testutil.Ok(t, err)
+	d.client.BaseURL = endpoint
+
+	ctx := context.Background()
+	tgs, err := d.refresh(ctx)
+	testutil.Ok(t, err)
+
+	testutil.Equals(t, 1, len(tgs))
+
+	tg := tgs[0]
+	testutil.Assert(t, tg != nil, "tg should not be nil")
+	testutil.Assert(t, tg.Targets != nil, "tg.targets should not be nil")
+	testutil.Equals(t, 2, len(tg.Targets))
+
+	for i, lbls := range []model.LabelSet{
+		{
+			"__address__":            model.LabelValue("104.236.32.182:80"),
+			"__meta_do_id":           model.LabelValue("3164444"),
+			"__meta_do_name":         model.LabelValue("example.com"),
+			"__meta_do_image":        model.LabelValue("ubuntu-16-04-x64"),
+			"__meta_do_private_ipv4": model.LabelValue(""),
```

Comment:
@brian-brazil is this OK or should I not add the empty label?

Reply:
It should be harmless, if not that's a bug in the general SD code.